File: C/Exponential_Search.c
int exponentialSearch(int arr[], int n, int key) {
    if (arr[0] == key)
        return 0;
    int i = 1;
    while (i < n && arr[i] <= key)
        i *= 2;
    return binarySearch(arr, i/2, min(i, n-1), key);
}

File: Python/Exponential_Search.py
def exponential_search(arr, key):
    if arr[0] == key:
        return 0
    n = len(arr)
    i = 1
    while i < n and arr[i] <= key:
        i *= 2
    return binary_search(arr[min(i, n-1):i//2-1:-1], key)
